app/services/concerns/spot/retrieves_s3_urls.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+#
+# app/services/concerns/spot/retrieves_s3_urls.rb
+module Spot
+  # Mixin to add to BrowseEverything::Retriever for
+  # retrieving `s3://` urls. BrowseEverything takes a configuration
+  # option for S3 url :response_type, but Retriever doesn't know
+  # how to handle them.
+  #
+  # This should be `.prepend` ed into the BrowseEverything::Retriever class
+  # in an initializer.
+  #
+  # @example
+  #   BrowseEverything::Retriever.prepend(Spot::RetrievesS3Urls)
+  #
+  module RetrievesS3Urls
+    extend ActiveSupport::Concern
+
+    module ClassMethods
+      def can_retrieve?(uri, _headers = {})
+        uri_parsed = ::Addressable::URI.parse(uri)
+
+        case uri_parsed.scheme
+        when "s3"
+          client = Aws::S3::Client.new
+          resp = client.head_object(bucket: uri_parsed.host, key: uri_parsed.path)
+          return true unless resp.nil?
+          return false
+        else
+          super(uri, _headers = {})
```

Comment:
I think this is assigning the `_headers` param to an empty hash before calling super; you probably just want `super(uri, _headers)`